apps/storefront/src/pages/ShoppingListDetails/components/QuickAdd.tsx

```diff
@@ -163,7 +166,12 @@ export default function QuickAdd(props: AddToListContentProps) {
 
         const quantity = (skuValue[sku] as number) || 0;
 
-        if (purchasingDisabled === '1' && type !== 'shoppingList') {
+        if (purchasingDisabled === '1' && !['shoppingList', 'quoteDraft'].includes(type || '')) {
+          notPurchaseSku.push(sku);
+          return;
+        }
+
+        if (purchasingDisabled === '1' && type === 'quoteDraft' && !allowAddNonPurchasableProduct) {
           notPurchaseSku.push(sku);
           return;
         }
```

Comment:
After the changes in this PR, the only two possible values for `type` are `shoppingList` and `quoteDraft`. There's no case where the value is neither.

<img width="505" height="320" alt="Image" src="https://github.com/user-attachments/assets/f2de00c9-e475-4369-9fad-1a75ce9d4ea8" />

Can we adjust the types to require `type: 'shoppingList' | 'quoteDraft'` and adjust this logic accordingly?